src/transformers/models/bigscience176b/modeling_bigscience176b.py

```diff
@@ -262,46 +290,83 @@ def forward(
         # Output. [sq, b, h]
         # =================
 
-        output = self.dense(context_layer)
+        # aggregate results across tp ranks
+        slices = context_layer.shape[-1] / self.pretraining_tp
+        output_tensor = torch.zeros_like(context_layer)
+        for i in range(self.pretraining_tp):
+            output_tensor = output_tensor + F.linear(context_layer[:, :, int(i*slices):int((i+1)*slices)], self.dense.weight[:, int(i*slices):int((i+1)*slices)])
+        if not self.skip_bias_add:
+            output_tensor = output_tensor + self.dense.bias if self.dense.bias is not None else output_tensor
+            output_bias = None
+        else:
+            output_tensor = output_tensor
+            output_bias = self.dense.bias
+        output = output_tensor
+        # output = self.dense(context_layer)
 
         outputs = (output, present)
         if output_attentions:
             outputs += (None,) # TODO not implemented yet
 
-        return outputs  # a, present, (attentions)
+        return outputs, output_bias  # a, present, (attentions)
 
 
 # Copied from transformers.models.gpt2.modeling_gpt2.GPT2MLP with GPT2->BigScience176B
```

Comment:
This "Copied from" line is actually a macro that ask the test in the transformers repo to make sure that the class we copied the original model from (`GPT2MLP`) and the class we have now (`BigScience176BMLP`) are identical and otherwise raise an error (you can ask Sylvain Gugger about it). We use it to make sure model stay in sync when they need to across evolution of the repo.

You should probably remove it here since we modified the class and don't expect it to stay identical

Reply:
Perfect ! I was not aware of that macro (which is actually quite useful for some cases I agree!) should be fixed [here](https://github.com/thomwolf/transformers/pull/1/commits/e27e554e15da432d0d0aafb60b554d7c9640bfa9)